drivers/char/tpm/tpm_tis_spi_main.c

```diff
@@ -37,6 +38,7 @@
 #include "tpm_tis_spi.h"
 
 #define MAX_SPI_FRAMESIZE 64
+#define SLB9670_RESET_DELAY	60 /* ms */
```

Comment:
The 60 ms are not the reset delay. It is the time the device must be running until the next reset can be asserted. Also the device reset looks a bit more complex then just asserting an deasserting the reset.
The Figure 3 RST# Timing shows a sequence:
1. deassert the RST# for 60ms
2. assert the RST# for 2us
3. deassert the RST# for 60ms
4. assert the RST# 2us
5. deassert the RST# for 60ms
6. TPM commands

I ignored the tPOR as it is unlikely we will reach the TPM driver in < 80us.

Reply:
Is this really the reset sequence? Or just an example for how to use the time intervals?

Reply:
In Section 5.5 of the data sheet it stats it has counter measures for attacks which also include reset behavior.
For example this quote:

> After the deassertion of the RST# signal, the system should wait for a minimum time of t RSTIN before asserting RST# again

I read it as that the reset sequence is not an example but mandatory.